### PR TITLE
Bring composer into the main OpenEMR codebase

### PR DESCRIPTION
**Pull request by Wakie87**

### CODE REVIEW

**Comment by bradymiller** [comment]
This PR is currently on hold.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/sites/default/sqlconf.php
```
Comment (bradymiller) [inline .gitignore]: I think this will break upgrading

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+/sites/default/sqlconf.php 
+/interface/main/calendar/modules/PostCalendar/pntemplates/
```
Comment (bradymiller) [inline .gitignore]: I am pretty sure this will break things (is this or above related to composer stuff?)

Comment (Wakie87) [inline .gitignore]: These are more for developers, I guess we can add something to the wiki about adding these into the global .gitignore file.

File: interface/globals.php [hunk 3]
```diff
@@ -147,6 +147,8 @@ function undoMagicQuotes($array, $topLevel=true) {
 // Static assets directory, relative to the webserver root.
 // (it is very likely that this path will be changed in the future))
 $GLOBALS['assets_static_relative'] = "$web_root/public/assets";
+//Composer vendor directory, relative to the webserver root.
+$GLOBALS['vendor_dir'] = "$web_root/vendor";
```
Comment (bradymiller) [inline interface/globals.php]: this is only a relative path. will need to use an absolute path:
``````
 $GLOBALS['vendor_dir'] = "$webserver_root/vendor";

File: interface/globals.php [hunk 4]
```diff
@@ -172,6 +174,9 @@ function undoMagicQuotes($array, $topLevel=true) {
 // Includes functions for date internationalization
 include_once (dirname(__FILE__) . "/../library/date_functions.php");
 
+// Includes compoaser autoload
+include_once (dirname(__FILE__) . "/../vendor/autoload.php");
```
Comment (bradymiller) [inline interface/globals.php]: shouldn't this be:
``````
require_once ($GLOBALS['vendor_dir'] ."/autoload.php");

Comment (bradymiller) [inline interface/globals.php]: As an aside, a good fix here would be to convert the two above include_once to require_once .

**Comment by Wakie87** [comment]
Updated with regards to previous comments.

**Comment by bradymiller** [comment]
Just committed this to the codebase.
thanks for the contribution!
-brady
